plugin/src/main/java/com/yelp/codegen/KotlinGenerator.kt

```diff
@@ -142,10 +142,27 @@ class KotlinGenerator : SharedCodegen() {
     public override fun listTypeWrapper(listType: String, innerType: String) =
             "$listType<$innerType>"
 
+    public override fun listTypeUnwrapper(baseType: String) = baseType
+            .replace(" ", "")
+            .removeSuffix(">")
+            .removePrefix("List<")
+            .removePrefix("Array<")
```

Comment:
As far as I can understand from the code the method will ever be called with `baseType` set to the type that will be output-ed on the file.
If the understanding is correct, then `Array<` should never be present

Reply:
Yup, you're right. Let's refactor this out :)